### ISSUE TITLE
Squash message doesn't behave like commit message (lines starting with # are silently removed)

### ISSUE THREAD

**Issue by FrancescElies** [desktop/desktop#16042]
### The problem

Squash message doesn't behave like commit message (lines starting with # are silently removed) this is a confusing behavior that most likely one might might not even notice.

 
Doing a rebase from the commandline treats lines starting with `#` as comments and in that context is obvious that those lines will be removed, I suspect the contents of the squash window message is forwarded raw to git rebase command, most of our colleagues using github desktop they don't feel comfortable with the commandline and might face this problem rather often as for tracking purposes internally we use e.g. `#0123456789` as a way to link back to work items.

### Release version

3.1.5 (x64)

### Operating system

Windows

### Steps to reproduce the behavior

1. ![grafik](https://user-images.githubusercontent.com/6558089/215773417-94eaa5ec-990d-47f4-99e1-5c7405c421ef.png)
1. ![grafik](https://user-images.githubusercontent.com/6558089/215773652-de0698a7-45ea-4526-941b-96f4f8a91bc5.png)
1. ![grafik](https://user-images.githubusercontent.com/6558089/215773822-2d5d1f73-060d-4d8a-bb8f-a68ce9d89278.png)
1. ![grafik](https://user-images.githubusercontent.com/6558089/215773920-6ec55fe4-dddc-43b2-ab77-3a45435fd81c.png)
1. ![grafik](https://user-images.githubusercontent.com/6558089/215773976-e6c98443-1256-40dc-8c97-0de35dd5418a.png)
  As you can see the squashed commit is missing lines that started with `#`

**Comment by philipplentzen** [comment]
Ran into this problem multiple times now, this is especially annoying when the first line of the commit message is the reference to an issue on GitHub as they are starting with a #. Now, every time I do a squash commit, I need to amend the commit message afterwards to include the issue number.

@sergiou87 any chance or plans to give this a higher priority?

**Comment by tidy-dev** [comment]
@philipplentzen This has not been prioritized. However, if you would be interested in taking a look at it. I suspect you are right and [this appending of commit summaries](https://github.com/desktop/desktop/blob/7a7d602a5f63711364c0042c0fbed3f01639eb2f/app/src/lib/git/squash.ts#L80) needs to be sanitized for '#' or the summary/description encapsulated in quotation marks. 

Warning that the teams bandwidth is small so we cannot guarantee prompt review of PR's. Hopefully this would be small PR tho.

**Comment by philipplentzen** [comment]
@tidy-dev Did some more investigation into this.

TLDR; It is the default git behavior for an interactive rebase, which cannot be escaped. The reference to issues is treated as a comment inside the message. For me setting `git config --global core.commentChar auto` did the trick. With that config, git detects the commenChar automatically inside commit messages.

The squash message defined inside the dialog and used as the commit message after the interactive rebase is set [here](https://github.com/desktop/desktop/blob/7a7d602a5f63711364c0042c0fbed3f01639eb2f/app/src/lib/git/squash.ts#L139-L142). The code lines you provided are part of the first (setting `pick` and `squash`inside the editor) interactive rebase step. 

The problem with providing the message after the interactive rebase is that the, by a # at the beginning of the line, referenced commits are interpreted as comments. The only suitable solution to this problem seems to be changing the `commentChar` inside the global git config. Found this [discussion](https://stackoverflow.com/a/14931661) where the introduction of an `auto` commentChar which will automatically detect the commentChar inside messages. Maybe this option could be something that is added to the Git section inside the settings dialog.

**Comment by tidy-dev** [comment]
Another possible solution would be to detect if one of the commits starts (on the start of an interactive rebase session) with `#` and then ask the user if they would like to set that git config. This is a pattern we have used elsewhere.

**Comment by philipplentzen** [comment]
Is this something you and the team would be interested in developing? Maybe by the community?

**Comment by steveward** [comment]
@philipplentzen you'd be welcome to work on this and submit a PR for us to take a look at!
